utilities/src/bits.rs

```diff
@@ -146,6 +146,14 @@ macro_rules! impl_bits_for_integer {
             /// Reads `Self` from a boolean array in little-endian order.
             #[inline]
             fn from_bits_le(bits: &[bool]) -> Result<Self> {
+                // If the number of bits exceeds the size of the integer, ensure that the upper bits are all zero.
+                // Note that because the input bits are little-endian, these are the bits at the end of slice.
+                if bits.len() > (<$int>::BITS as usize) {
+                    for bit in bits[<$int>::BITS as usize..].iter() {
+                        ensure!(!bit, "upper bits are not zero");
+                    }
+                }
```

Comment:
A `skip` call should replicate this logic and get rid of the `if` statement.

```suggestion
                for bit in bits.iter().skip(<$int>::BITS as usize) {
                    ensure!(!bit, "upper bits are not zero");
                }
```